### ISSUE TITLE
Refactor tests

### ISSUE THREAD

**Issue by malcolm-kee** [webpack/webpack-dev-server#4846]
### Bug report

This is to track the bad tests so I will remember to pick this up later.

Right now some tests will hangs when error, because the puppeteer does not cleanup correctly, such as tests in this file: https://github.com/webpack/webpack-dev-server/blob/master/test/e2e/overlay.test.js

We should include some cleanup if the puppeteer is not cleanup due to error.

**Comment by alexander-akait** [comment]
Yeah, maybe we can merge it with https://github.com/webpack/webpack-dev-server/issues/2843

**Comment by malcolm-kee** [comment]
I think #2843 would probably need more time, I remembered I was blocked by something related to websocket few weeks back.

Besides, this refactoring itself is already very big 😅
